### PR TITLE
fileNameWithPrefix and fileNameWithPostfix

### PR DESCRIPTION
**Pull request by hovosanoyan**
Functions which will let to add prefix or postfix to the file name in the path.

### CODE REVIEW

**Comment by RyanZim** [comment]
@jprichardson What's your thoughts on this? IDC

If this is going to be merged, it will need docs, tests, and also the actual implementation could be simplified. Will review in greater detail if we approve adding this feature.

**Comment by jprichardson** [comment]
> @jprichardson What's your thoughts on this? IDC

IDC either.

> If this is going to be merged, it will need docs, tests, and also the actual implementation could be simplified. Will review in greater detail if we approve adding this feature.

Agreed 100%

What's the real-world use case @hovosanoyan?

**Comment by hovosanoyan** [comment]
> What's the real-world use case @hovosanoyan?

When working with files, in my case I often need to copy file with different name. 
example 1. From video file get separated audio and video, so it will be easy to quickly name it
filename-audio.mp3, filename-video.mp4.

example 2. I need to copy files, so i can easily rename them like this
filename (1).mp4, filename (2)mp4

example 3. Create copy of file and prefix with date. like this
dd-mm-yy-filename.mp4

And there is no need to work with file names when you just need the path. In my code i prefer to use absolute path in functions instead of file names. So i think this will be useful for renaming file using path. 
I'm not sure how often others needs this, so want to listen what you think about this.

**Comment by RyanZim** [comment]
@jprichardson What's your decision on this?

**Comment by jprichardson** [comment]
> @jprichardson What's your decision on this?f

IF neither of us care, I certainly don't, then it's fine with we accept given it meets our conditions of tests and documentation.

File: lib/path.js [hunk 1]
```diff
@@ -15,6 +15,21 @@ path.normalizeTrim = function (str) {
   return path.normalize(str).replace(new RegExp(escapeRegexp(path.sep) + '$'), '')
 }
 
+
+path.fileNameWithPostfix = function (filename, postfix) {
+  var ext = path.extname(filename)
+  var fileNameWithoutExt = path.basename(filename).replace(ext, '')
```
Comment (RyanZim) [inline lib/path.js]: Should be:

    path.basename(filename, ext)

File: lib/path.js [hunk 2]
```diff
@@ -15,6 +15,21 @@ path.normalizeTrim = function (str) {
...
+  var output = path.join(path.dirname(filename), fileNameWithoutExt + postfix + ext)
+  return output
+}
+
+path.fileNameWithPrefix = function (filename, prefix) {
+  var ext = path.extname(filename)
+  var fileNameWithoutExt = path.basename(filename).replace(ext, '')
```
Comment (RyanZim) [inline lib/path.js]: Same here

**Comment by hovosanoyan** [comment]
Ok, I think current version have everything is required.

**Comment by RyanZim** [comment]
@hovosanoyan Looks pretty good, but now you have two implementations in `lib/path.js`; the old one and the new one. You gotta delete the old implementation.

**Comment by RyanZim** [comment]
Will merge and release when I get a chance

**Comment by RyanZim** [comment]
Published in 4.2.0

**Pull request merged**